app/models/cart.rb

```diff
@@ -0,0 +1,19 @@
+class Cart < ApplicationRecord
+  belongs_to :user
+
+  has_many :cart_items, dependent: :destroy
+
+  def add_cart_item(product, quantity)
+    cart_item = cart_items.find_or_create_by(product:)
```

Comment:
find_or_create_by だと下の save! と合わせて2回SQLが実行されるので、find_or_initialize_by かなと思います
```suggestion
    cart_item = cart_items.find_or_initialize_by(product:)
```

Reply:
なんと…意識せずに使ってました…
[1622e5d](https://github.com/kk0000-kk/sakura-market/pull/25/commits/1622e5d227dc49c67246678d8784c71dfb35792e)

そして、バグを見つけました
[e8b4b5c](https://github.com/kk0000-kk/sakura-market/pull/25/commits/e8b4b5c48059d6f4ebc48cb47b28efef6b0e9435)